Tab logic refactored

gemdev111 opened the pull request:
*Created NavigationStateManager
- Moved navigation pathes & tabs handling logic from coordinator to NavigationStateManager
- Refactored logic of handling tabs
- Added logic for handling scroll to top when tab selected 

*Added NavigationStateManager in WalletCoordinator
- Removed unnecessary navigation dependencies from coordinator
- Added MainTabView to separate the view and tab states
- Moved TabItem from Wallet module to Types in Coordinator module

Finished #4

gemdev111 committed:
Tab logic refactored

*Created NavigationStateManager
- Moved navigation pathes & tabs handling logic from coordinator to NavigationStateManager
- Refactored logic of handling tabs
- Added logic for handling scroll to top when tab selected 

*Added NavigationStateManager in WalletCoordinator
- Removed unnecessary navigation dependencies from coordinator
- Added new
- Moved TabItem from Wallet module to Types in Coordinator module

Finished #4

gemdev111 committed:
Updated navigationStateManager init to compatible with Github CI swift-linter
git submodules updated

gemcoder21 commented on Gem/Core/Services/NavigationStateManager.swift:
```diff
@@ -0,0 +1,86 @@
+typealias TabScrollToTopId = TabItem
+
+protocol NavigationStateManagable: Observable {
+    var walletNavigationPath: NavigationPath { get set }
```
```suggestion
    var wallet: NavigationPath { get set }
```

to simplify naming, as it's called `NavigationStateManager`

gemcoder21 commented on Gem/Settings/Scenes/SettingsScene.swift:
```diff
@@ -36,6 +36,8 @@ struct SettingsScene: View {
                     ListItemView(title: Localized.Settings.security, subtitle: .none, image: Image(.settingsSecurity))
                 }
             }
+            .id(TabScrollToTopId.settings)
```
is it possible to setup `.id()` from the outside of SettingsScene? in navigation coordinator? to avoid having too much code inside each screen

gemdev111 replied:
removed since, it was request to remove it by now

gemdev111 committed:
TabView adjustements

- Added @Bindable TODO
- Moved logic of selection tab in a tabview
- Added serveral functions for handling navigation and tab selection to a manager
- Removed unused code related to scrolling to top

gemdev111 committed:
Merge branch 'main' into tabbar-scroll-to-top
